### ISSUE TITLE
Cannot read property 'timeout' of undefined and crash

### ISSUE THREAD

**Issue by VerardiA** [node-red/node-red#2471]
### What are the steps to reproduce?

    Run Node Red on Portainer or Docker.
    Add Mqtt Input node, some delay nodes and trigger nodes.
    Inject some flow into it and deploy.

### What happens?

NodeRed completely crashes after a while.

### What do you expect to happen?

NodeRed to report an error to fix.

### Please tell us about your environment:

- [ ] Node-RED version: v1.0.3
- [ ] Node.js version: v10.18.0
- [ ] npm version: Edge Gateway NIOT-E-TPI51-EN-RE
- [ ] Platform/OS:
- [ ] Browser: Firefox

### Docker Logs:
    20 Feb 21:30:27 - [warn] Communication send error: Error: WebSocket is not open: readyState 3      (CLOSED)

    20 Feb 22:00:09 - [red] Uncaught Exception:

    20 Feb 22:00:09 - TypeError: Cannot read property 'timeout' of undefined

    at completeSend (/usr/lib/node_modules/node-red/node_modules/@node-red/nodes/core/sequence/17-split.js:432:32)

    at Timeout._onTimeout (/usr/lib/node_modules/node-red/node_modules/@node-red/nodes/core/sequence/17-split.js:624:29)

    at ontimeout (timers.js:436:11)

    at tryOnTimeout (timers.js:300:5)

    at listOnTimeout (timers.js:263:5)

    at Timer.processTimers (timers.js:223:10)

### Fix:

In order to get NodeRed running again, I had to restart the docker container manually.

**Comment by knolleary** [comment]
Can you provide the flow that caused this crash?

**Comment by VerardiA** [comment]
Sorry i can't provide the flow because the device is installed on my client network and there are sensitive data about his company inside. Sorry for my english by the way.

**Comment by knolleary** [comment]
We need more details to understand what is happening. If you cannot provide the actual flow, can you provide a simplified flow that recreates it? Or maybe a better description of what nodes are involved, how they are configured and what the messages look like.

In particular, what messages are being passed to any Split node you have

**Comment by VerardiA** [comment]
Ok, i've recreated the flow without credentials

[flows(5).zip](https://github.com/node-red/node-red/files/4235088/flows.5.zip)

**Comment by knolleary** [comment]
@VerardiA that flow has over 400 nodes in it - many of which I don't have installed locally. Please can you narrow it down to just the flow that is hitting the issue.

What we really need to know is:
1. how is the Split node configured?
2. what message(s) are you passing it?

Ideally, this would be just an Inject/Change node pair to create the message that is then passed to a Split node to recreate the issue.

**Comment by VerardiA** [comment]
After a more in-depth search maybe I found the problem.

There is a join node that combines 92 modbus messages from the plc, the "completeSend" error is generated by the join node and not the split node.

I tried to remove the join node timeout and replace it with a trigger node. If the problem persists I will let you know.

Thanks for pointing out that the join and split nodes come from the same file.

**Comment by HenryNe** [comment]
[flow-join-crash-20200623.json.txt](https://github.com/node-red/node-red/files/4820592/flow-join-crash-20200623.json.txt)
This flow triggers the crash 3 second after start.
It's a timestamp inject, a function to create the `msg.complete`, and a join node with Mode "_manual_", _key/value as Object_. The only trigger is waiting for "_After a message with the msg.complete property set_".

This is the backtrace:
```
TypeError: Cannot read property 'timeout' of undefined
    at completeSend (/usr/local/node-red/nodes/core/logic/17-split.js:432:23)
    at JoinNode.<anonymous> (/usr/local/node-red/nodes/core/logic/17-split.js:542:29)
    at emitOne (events.js:116:13)
    at JoinNode.emit (events.js:211:7)
    at JoinNode.Node.receive (/usr/local/node-red/red/runtime/nodes/Node.js:215:14)
    at FunctionNode.Node.send (/usr/local/node-red/red/runtime/nodes/Node.js:202:14)
    at sendResults (/usr/local/node-red/nodes/core/core/80-function.js:52:18)
    at FunctionNode.<anonymous> (/usr/local/node-red/nodes/core/core/80-function.js:221:21)
    at emitOne (events.js:116:13)
    at FunctionNode.emit (events.js:211:7)
    at FunctionNode.Node.receive (/usr/local/node-red/red/runtime/nodes/Node.js:215:14)
    at InjectNode.Node.send (/usr/local/node-red/red/runtime/nodes/Node.js:137:22)
    at InjectNode.<anonymous> (/usr/local/node-red/nodes/core/core/20-inject.js:77:22)
    at emitOne (events.js:116:13)
    at InjectNode.emit (events.js:211:7)
    at Timeout._onTimeout (/usr/local/node-red/nodes/core/core/20-inject.js:46:20)
```
Versions:

> Node-RED version: v0.18.7
> Node.js  version: v8.11.3
> 

The original line was: [17-split.js#L432](https://github.com/node-red/node-red/blob/472bbdb59f59108b7fa8624f1a8a2110d213530c/nodes/core/logic/17-split.js#L432)

Current version of this line also not helps:
https://github.com/node-red/node-red/blob/708620f929199b392c5a30cae502b540f94546d0/packages/node_modules/%40node-red/nodes/core/sequence/17-split.js#L430

With this lines found, that "group" is undefined:
```
if (group !== undefined) {
    if (group.timeout !== undefined) {
        console.log("group.timeout is OK!");
    } else {
        console.log("group.timeout is UNDEFINED!");
    }
} else {
    console.log("group is UNDEFINED!");
}
console.log( 'group: ' + JSON.stringify(group) );
if (group.timeout) { clearTimeout(group.timeout); }
```
Output:

> group is UNDEFINED!
> group: undefined
> TypeError: Cannot read property 'timeout' of undefined
>

**Comment by knolleary** [comment]
Fixed in the `dev` branch for the 1.1.0 release.
